server.go

```diff
@@ -345,6 +345,18 @@ func newServer(listenAddrs []net.Addr, chanDB *channeldb.DB, cc *chainControl,
 		ExtractErrorEncrypter:  s.sphinx.ExtractErrorEncrypter,
 		FetchLastChannelUpdate: s.fetchLastChanUpdate(),
 		Notifier:               s.cc.chainNotifier,
+		NotifyActiveChannelEvent: func(chanID lnwire.ShortChannelID) {
+			select {
+			case s.channelNotifier.NewActiveChannels.ChanIn() <- chanID:
+			case <-s.quit:
+			}
+		},
+		NotifyInactiveChannelEvent: func(chanID lnwire.ShortChannelID) {
+			select {
+			case s.channelNotifier.NewInactiveChannels.ChanIn() <- chanID:
```

Comment:
To make sure is always non-blocking, I'm wondering if we should select on `default` here? Or somehow determine that the queue is active somehow.

Need to think about this abit.

Reply:
let me know if you like my solution to this!